gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -420,5 +420,191 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
     }
   }
 
-  // TODO(mderka) Add methods. Created issue #596.
+  /**
+   * Creates a new zone.
+   *
+   * @return ZoneInfo object representing the new zone's metadata. In addition to the name, dns name
+   * and description (supplied by the user within the {@code zoneInfo} parameter, the returned
+   * object will include the following read-only fields supplied by the server: creation time, id,
+   * and list of name servers.
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/create">Cloud DNS Managed Zones:
+   * create</a>
+   */
+  ZoneInfo create(ZoneInfo zoneInfo);
+
+  /**
+   * Retrieves the zone by the specified zone name. Returns {@code null} is the zone is not found.
+   * The returned fields can be optionally restricted by specifying {@code ZoneFieldOptions}.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/get">Cloud DNS Managed Zones:
+   * get</a>
+   */
+  ZoneInfo getZone(String zoneName, ZoneOption... options);
+
+  /**
+   * Retrieves the zone by the specified zone name. Returns {@code null} is the zone is not found.
+   * The returned fields can be optionally restricted by specifying {@code ZoneFieldOptions}.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/managedZones/get">Cloud DNS Managed Zones:
+   * get</a>
+   */
+  ZoneInfo getZone(BigInteger zoneId, ZoneOption... options);
```

Comment:
I see your point but I do not agree. Your arguments makes sense in this context but is dangerous in general. We cannot design a new and more usable library that also addresses all previous usages. The question is: providing both ways is a good design choice? Is it maintainable in the long term? Does it create confusion? Are users going to benefit from this?

Also regarding the service it would be nice to know why are both id and name supported.